static/js/lib/forms.js

```diff
@@ -0,0 +1,306 @@
+// Submit a form with ajax, call as:
```

Comment:
The code in this file is reusable across many forms and ideally will be documented in the developer docs.